src/Form/Form.js

```diff
@@ -3,27 +3,54 @@ import classes from "./Form.module.css";
 import * as firebase from "../lib/firebase";
 
 const Form = props => {
+  const { setdbItems, dbItems, token, onEnterToken } = props;
   const [name, setName] = useState("");
   const [frequency, setFrequency] = useState(0);
   const [date, setDate] = useState("");
 
   const handleSubmit = e => {
     e.preventDefault();
     let db = firebase.fb.firestore();
-    let data = {
-      name,
-      frequency,
-      date
-    };
-    db.collection(props.token)
-      .add(data)
-      .then(function() {
-        alert("Item added!");
-      })
-      .catch(function(error) {
-        alert("Something went wrong!");
-        console.error("Error writing document: ", error);
-      });
+    let newName = name
+      .toUpperCase()
+      .replace(/[^\w\s]|_/g, "")
+      .replace(/\s+/g, " ");
```

Comment:
Nice use of Regex!

On first glance, though, I can't understand why there are two different lines for regex. I happen to know how regex works, so I see what you're doing, but your teammates might not. I suggest adding a comment above each regex replacement stating exactly what it's doing.

Reply:
I agree with @rmorabia here 💯 

I've been writing regex since 2006 and I still need to study them to figure out what's going on 😅 A comment would be really helpful.

Reply:
I would agree with the above comments, I've seen super long regex checks on one line and can never remember all the ins and outs of those checks.

Reply:
Thanks! I'm still working out when/where to leave comments so this is really helpful.